code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -404,41 +403,58 @@
 
 	var/obj/item/I = absorber.item
 
-	if(I)
-		var/absorb = clamp(src.absorb_rate, 0, I.health)
+	if (!I)
+		return
+
+	var/absorb = min(src.absorb_rate, I.health)
+	if (absorber.instant_absorb && !absorber.ignore_amount)
+		boutput(src, "<span class='alert'>[I] is weak enough that it breaks apart instantly!</span>")
+		src.resources += round(src.absorb_per_health * absorb * I.amount)
```

Comment:
However improbable, couldn't this lead to an entire stack being absorbed instantly if its current health is less than 1?

Reply:
Yeah, that's why it's included, but it's for resources gained for a single item in the stack being zero. It's to prevent cases like if you were to get zero resources per item in a stack, basically you'd spend the entire time dissolving the item without gaining any resources at all.